storeinterface/storeinterface.go

```diff
@@ -0,0 +1,145 @@
+package storeinterface
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"time"
+)
+
+var retryScaler = 1.0
+
+const maxRetries = 6
+
+// StoreInterface is able to upload the contents of a Reader to the SD Store
+type StoreInterface interface {
+	Upload(path string, filePath string) error
+}
+
+type sdUploader struct {
+	url    string
+	token  string
+	client *http.Client
+}
+
+// NewFileUploader returns an StoreInterface for a given url.
+func NewFileUploader(url, token string) StoreInterface {
+	return &sdUploader{
+		url,
+		token,
+		&http.Client{Timeout: 30 * time.Second},
```

Comment:
not sure if we need all this?
I already integrated the httpClient into the `store-cli.go` itself. Don't think we need this whole sdUploader?

Reply:
`store-cli.go` doesn't seem to have the logic to upload the contents of a file so far. It's also not using the `val` parameter you pass to the `set` function. Have you pushed all your changes?